lib/peep/codegen.ex

```diff
@@ -39,33 +42,35 @@ defmodule Peep.Codegen do
     quote do
       def handle_event(event, measurements, metadata, _) do
         global_tags = global_tags()
-        global_tags_keys = global_tags_keys()
 
         %Persistent{
           events_to_metrics: %{^event => metrics},
           storage: {storage_mod, storage}
-        } = Persistent.fetch(unquote(peep_name))
-
-        for {metric, id} <- metrics do
-          %{
-            measurement: measurement,
-            tag_values: tag_values,
-            tags: tags,
-            keep: keep
-          } = metric
-
-          case keep?(keep, metadata) && fetch_measurement(measurement, measurements, metadata) do
-            value when is_number(value) ->
-              tag_values = Map.merge(global_tags, tag_values.(metadata))
-              tags = Map.new(global_tags_keys ++ tags, &{&1, Map.get(tag_values, &1, "")})
-              storage_mod.insert_metric(storage, id, metric, value, tags)
-
-            _ ->
-              nil
-          end
-        end
-
-        :ok
+        } = Persistent.fast_fetch(unquote(peep_name))
+
+        :lists.foreach(
+          fn {metric, id} ->
+            %{
+              measurement: measurement,
+              tag_values: tag_values,
+              tags: tags,
+              keep: keep
+            } = metric
+
+            if keep?(keep, metadata) do
+              case fetch_measurement(measurement, measurements, metadata) do
```

Comment:
Why was the &&/2 removed?

Splitting these shouldn't have any performance impact, as `Kernel.&&/2` won't evaluate fetch_measurement/3 if keep?/2 returns false. Classic example:

```elixir
iex> false && throw(:bad)
false
```

Credo does begin complaining about nesting with this change. We could keep it this way, and add `# credo:disable-for-next-line Credo.Check.Refactor.Nesting` above the case expression.

Reply:
I removed it, as is IMHO more readable than `&&`, especially there.